ui/src/features/get-tests.js

```diff
@@ -60,6 +69,7 @@ class getTests extends React.Component {
     if (prevProps.tests !== this.props.tests) {
       this.setState({ sortedTests: [...this.props.tests], sortHeader: 'updated_at-' }, () => {
         this.onSort('updated_at');
+        this.filterByFavoriteState();
```

Comment:
First of all excellent PR!! thanks!!

Here we have a problem because we have here two async set state, which we don't know what will be first, so we might lose the sort state.

as a solution I would add on componentDidupdate if sortedTests has changed -> run filter by favorite test unless you have other solution 

again,
thank you for the contribution for that project! its awesome!!

@enudler - do some qa as you like:)

Reply:
@manorlh - I couldn't do the above as `filterByFavoriteState` also causes a change in `state.sortedTests` and calling on state change would cause an infinite loop. Instead I moved the sort call to `filterByFavoriteState` as there wasn't any particular reason why not to sort it by updated time.